hts.c

```diff
@@ -938,10 +938,15 @@ htsFile *hts_open_format(const char *fn, const char *mode, const htsFormat *fmt)
          fmt->format == fastq_format))
         fp->format.format = fmt->format;
 
-    if (fmt && fmt->specific)
-        if (hts_opt_apply(fp, fmt->specific) != 0)
+    if (fmt && fmt->specific) {
+        if (hts_opt_apply(fp, fmt->specific) != 0) {
+            if (((hts_opt*)fmt->specific)->opt == CRAM_OPT_REFERENCE) {
+                //reset error on reference file to avoid incorrect error report
+                errno = 0;
```

Comment:
Rather than reset `errno` unconditionally, it would be better to take a copy before calling `hts_opt_apply()` and then use that to restore the old value if necessary.  It might also be good to only do this if `errrno` is `ENOENT`, so we only suppress the "No such file or directory" errors and leave others alone.

E.g. something like this:
```
    if (fmt && fmt->specific) {
        int save_errno = errno;
        if (hts_opt_apply(fp, fmt->specific) != 0) {
            if (((hts_opt*)fmt->specific)->opt == CRAM_OPT_REFERENCE  && errno == ENOENT) {
                //reset error on reference file to avoid incorrect error report
                errno = save_errno;
```

Reply:
made changes as discussed.